src/layers/icmpv6.rs

```diff
@@ -176,28 +304,43 @@ impl Layer for ICMPv6 {
             }
 
             ICMPV6_ROUTER_ADVERTISEMENT => {
-                let mut flags = RouterAdvFlags::default();
                 let cur_hop_limit = u8::from_be(bytes[4]);
-                flags.managed_address_flag = ((bytes[5] >> 7) & 0x01) == 0x01;
-                flags.other_config = ((bytes[5] >> 6) & 0x01) == 0x01;
+                let flags = RouterAdvFlags {
+                    managed_address_flag: ((bytes[5] >> 7) & 0x01) == 0x01,
+                    other_config: ((bytes[5] >> 6) & 0x01) == 0x01,
+                };
 
                 let router_lifetime: u16 = (bytes[6] as u16) << 8 | (bytes[7] as u16);
                 let reachable_time: u32 = u32::from_be_bytes(bytes[8..12].try_into().unwrap());
                 let retrans_timer: u32 = u32::from_be_bytes(bytes[12..16].try_into().unwrap());
+
                 decoded = 16;
+                let options = handle_icmpv6_options(&bytes, &mut decoded);
```

Comment:
Here you can pass `&bytes[decoded..]`. This would then take care of the current offset. This is why I believe you were sending `decoded` value to the function call. 

This function call could look like - 
```rust

let (option_decoded, options) = handle_icmpv6_options(&bytes[decoded..])?; // Note the last `?` that would take care of the error part and the `Ok` value will be unpacked and matched with the left side. 

decoded += options_decoded;

```

Reply:
@gabhijit sir, I implemented the above changes suggested. The code looks much cleaner now. Thank you for the inputs.